libyara/modules/dotnet/dotnet.c

```diff
@@ -16,66 +16,1639 @@ limitations under the License.
 
 #include <ctype.h>
 #include <stdarg.h>
+#include <stdbool.h>
 #include <stdio.h>
+#include <string.h>
 #include <time.h>
 #include <yara/dotnet.h>
 #include <yara/mem.h>
 #include <yara/modules.h>
 #include <yara/pe.h>
 #include <yara/pe_utils.h>
+#include <yara/simple_str.h>
 #include <yara/strutils.h>
 
 #define MODULE_NAME dotnet
 
+static uint32_t max_rows(int count, ...)
+{
+  va_list ap;
+  int i;
+  uint32_t biggest;
+  uint32_t x;
+
+  if (count == 0)
+    return 0;
+
+  va_start(ap, count);
+  biggest = va_arg(ap, uint32_t);
+
+  for (i = 1; i < count; i++)
```

Comment:
Define `i` in the loop, like `for (int i = 1; ...)`. There are many places in the code were we already define these variables outside the `for` statement, this was because certain old Microsoft compilers didn't support it, but we are not using those compilers anymore.